core/src/test/java/kafka/log/remote/quota/RLMQuotaManagerTest.java

```diff
@@ -0,0 +1,118 @@
+        quotaManager.record(500);
+        // Move clock by 1 sec, quota is violated
+        moveClock(1);
+        assertTrue(quotaManager.isQuotaExceeded());
+
+        // Move clock by another 8 secs, quota is still violated for the window
+        moveClock(8);
+        assertTrue(quotaManager.isQuotaExceeded());
+
+        // Move clock by 1 sec, quota is no more violated
+        moveClock(1);
+        assertFalse(quotaManager.isQuotaExceeded());
+    }
+
+    @Test
+    public void testQuotaUpdate() {
+        RLMQuotaManager quotaManager = new RLMQuotaManager(
+            new RLMQuotaManagerConfig(50, 11, 1), metrics, QUOTA_TYPE, DESCRIPTION, time);
+
+        assertFalse(quotaManager.isQuotaExceeded());
+        quotaManager.record(51);
+        assertTrue(quotaManager.isQuotaExceeded());
+
+        Map<MetricName, KafkaMetric> fetchQuotaMetrics = metrics.metrics().entrySet().stream()
+            .filter(entry -> entry.getKey().name().equals("byte-rate") && entry.getKey().group().equals(QUOTA_TYPE.toString()))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        Map<MetricName, KafkaMetric> nonQuotaMetrics = metrics.metrics().entrySet().stream()
+            .filter(entry -> !entry.getKey().name().equals("byte-rate") || !entry.getKey().group().equals(QUOTA_TYPE.toString()))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        assertEquals(1, fetchQuotaMetrics.size());
+        assertFalse(nonQuotaMetrics.isEmpty());
+
+        Map<MetricName, MetricConfig> configForQuotaMetricsBeforeUpdate = extractMetricConfig(fetchQuotaMetrics);
+        Map<MetricName, MetricConfig> configForNonQuotaMetricsBeforeUpdate = extractMetricConfig(nonQuotaMetrics);
+
+        // Update quota to 60, quota is no more violated
+        Quota quota60Bytes = new Quota(60, true);
+        quotaManager.updateQuota(quota60Bytes);
+        assertFalse(quotaManager.isQuotaExceeded());
+
+        // Verify quota metrics were updated
+        Map<MetricName, MetricConfig> configForQuotaMetricsAfterFirstUpdate = extractMetricConfig(fetchQuotaMetrics);
+        assertNotEquals(configForQuotaMetricsBeforeUpdate, configForQuotaMetricsAfterFirstUpdate);
+        fetchQuotaMetrics.values().forEach(metric -> assertEquals(metric.config().quota(), quota60Bytes));
+        // Verify non quota metrics are unchanged
+        assertEquals(configForNonQuotaMetricsBeforeUpdate, extractMetricConfig(nonQuotaMetrics));
+
+        // Update quota to 40, quota is violated again
+        Quota quota40Bytes = new Quota(40, true);
+        quotaManager.updateQuota(quota40Bytes);
+        assertTrue(quotaManager.isQuotaExceeded());
+
+        // Verify quota metrics were updated
+        assertNotEquals(configForQuotaMetricsAfterFirstUpdate, extractMetricConfig(fetchQuotaMetrics));
+        fetchQuotaMetrics.values().forEach(metric -> assertEquals(metric.config().quota(), quota40Bytes));
```

Comment:
consider swapping:

assertEquals(actual, expected) to assertEquals(expected, actual)

Reply:
Yes, thanks.